### ISSUE TITLE
Syncthingtray not connecting to Syncthing after suspend

### ISSUE THREAD

**Issue by Nutomic** [Martchus/syncthingtray#16]
I am running syncthingtray on my laptop with Manjaro KDE. Syncthing is started via Autostart in the system settings (not systemd). Now whenever I wake my laptop from suspend, syncthingtray does not connect to it at all, i have to manually click the connect button. It would be great if this could work automatically, eg by having syncthingtray try to connect every 10 seconds (or some other interval).

Syncthing Tray, version 0.7.4
Syncthing v0.14.46

**Comment by Martchus** [comment]
I did some refactoring regarding that code anyways (because it already looked a bit faulty). So Maybe this is already fixed on latest master? I haven't tested this myself but I will before releasing 0.7.5. There are [PKGBUILDs](https://github.com/Martchus/PKGBUILDs) for the Git version if you like to test.

Note that a *delayed* reconnect after suspension is expected because the tray needs to give Syncthing some time to become available again. I guess this delay was 5 or 10 seconds. Since the disconnect happened due to suspension, the 'disconnected' notification is suppressed. If the delayed reconnect fails you should receive the usual 'can not connect' notification. Configuring a re-connect interval is already possible. So if you configured a reconnect interval, it should now try to re-establish the connection periodically. Seems like this didn't work in your case?

BTW: If the tray can not detect suspension (1) there's no delay and the reconnect interval is used directly. In this case the 'disconnected' notification can not be suppressed and should show up after suspension if enabled in the settings.  
(1) Detecting suspension only works for systemd and when you're user session D-Bus daemon is configured correctly to play well with systemd.

**Comment by Nutomic** [comment]
Ah you are right, I missed the reconnect setting, will give that a try. Maybe you could enable that by default?

**Comment by Martchus** [comment]
And, does it work?

The reconnect would be annoying if one actually want the service not to run always. Since the launcher and systemd unit status can now be considered to prevent connection attempts when Syncthing isn't running anyways this problem present anymore. So yes, I could make it default. Maybe 30 seconds or so.

**Comment by Martchus** [comment]
Good. I also set the auto-reconnect default to 30 seconds.

**Comment by stefs** [comment]
> I also set the auto-reconnect default to 30 seconds.

Hello, it seems that the auto-reconnect default is disabled again, and not 30 seconds:

https://github.com/Martchus/syncthingtray/blob/b5aa29c98b7b30b7e2ef0644708de1d3f041cf91/connector/syncthingconnectionsettings.h#L50

**Comment by Martchus** [comment]
Seems like it got lost when refactoring. I could change it back to 30 seconds if that's wanted.

**Comment by stefs** [comment]
I installed syncthingtray 1.1.10-1 from AUR on 5.10.68-1-MANJARO x86_64. I configured it with "Start the tray icon when the desktop environment launches" and "Launch Syncthing when starting the tray icon".
With this configuration out of the box syncthing did start as expected, but the UI did never connect automatically after startup. I had to click connect manually after every boot.
With the Reconnect set to 30 secs it connects after startup. However I don't know if my experience can be generalized for all setups.
